File: StringVeDiziOdevleri/Q34/main.c
#include <stdio.h>
#include <math.h>

int main()
{
    int n = 0;
    printf("sayi girin\n->");
    scanf("%d", &n);

    for (int i = 0; i < n; i++)
    {
        if (i == 0 || i == n - 1)
        {
            for (int j = 0; j < n / 2 + 1; j++)
                printf("*");

            printf("\n");
            continue;
        }

        printf("*");

        for (int j = 0; j < abs((n / 2) - i); j++)
            printf(" ");
        printf("*");

        printf("\n");
    }

    return 0;
}

File: StringVeDiziOdevleri/Q34/main.py
n = int(input("bir sayi girin\n->"))


def absF(x):
    return x if x > 0 else -x


for i in range(n):
    if i in [0, n - 1]:
        for _ in range(int(absF(n / 2 + 1))):
            print("*", end="")
        print("")
        continue

    print("*", end="")
    for _ in range(int(absF((n / 2) - i))):
        print(" ", end="")
    print("*", end="")
    print("")
